qdb/scripts/orchestrator.py

```diff
@@ -110,25 +111,35 @@ def cleanup_reports_dir(self):
             except FileNotFoundError:
                 logger.error(f"Could not delete from reports directory: {f}")
 
-    def get_recipients(self, unit_id, unit_name):
-        # recipients is initialized to either developers (dev mode) or to LBS (prod mode);
-        # unit recipients are added.
+    def get_recipients(self, unit_id: int, unit_name: str) -> set:
+        """Gets the recipients (email addresses) to which a unit's report
+        should be sent.
+        """
+
+        # This is set on class instantiation based on DEFAULT_RECIPIENTS,
+        # either to developers (dev mode) or to LBS (prod mode).
         recipients = set(self.recipients)
-        cmd = """
-            SELECT email
-            FROM qdb_staff, qdb_unit, qdb_recipient
-            WHERE qdb_recipient.unit_id = %s
-            AND qdb_staff.name != %s"""
+
+        # Legacy code alert: Apparently, if unit_name parameter is "LBS", only the
```

Comment:
I'm confused about this set of comments. Why doesn't this work in the web UI? It seems like when `run()` calls `get_recipients()`, if the unit is LBS, `unit_name` will be `LBS` and this logic will happen.

Reply:
It's true that if the user selects `LBS` as the unit in the web UI, this special logic runs.  I replicated the intent of the original query (full before & after context below).

My confusion (which led to the code comment): I think this is completely unnecessary.  LBS sets the recipients for each unit's report, via the Admin interface.  For the [LBS unit](http://127.0.0.1:8000/admin/qdb/recipient/?q=LBS) (sample data, but verified in production), only the Head is listed as a recipient.  With the web UI, the only way `unit_name == LBS` can occur is if LBS is the selected unit, so the data makes this moot.  (Discounting the possibility of someone hacking in, adding themselves as a non-Head recipient for the LBS report....)

Maybe when this was a command-line utility, run only by Josh, it was possible to pass non-matching unit id and unit name - for example, to run the Arts report by id, but with LBS as the unit name.  But even so, that would have sent the report only to the Arts head, not to LBS.

Then there's the [app-specific settings](https://github.com/UCLALibrary/LBS/blob/main/qdb/scripts/settings.py#L38-L56) which also factor in, by setting the `DEFAULT_RECIPIENTS` which get passed to `Orchestrator` on initialization.  In production, this is set to `LBS_RECIPIENTS` - the LBS Head and a non-head.  These people get a copy of each report, but apparently if LBS is the selected unit, only the LBS Head should get that report.

So, as much as I just want to rip out the `if...else` in question... I think it should stay.  But revisiting the code and writing this, I think the comment can be clearer, so I'll push a change shortly.

```
         if unit_name == "LBS":
-            cmd += " AND (qdb_staff.id = qdb_recipient.recipient_id"
-            cmd += " AND qdb_recipient.role = 'head')"
+            roles = ["head"]
         else:
-            cmd += """
-              AND ((qdb_staff.id = qdb_recipient.recipient_id AND qdb_recipient.role = 'head')
-                OR (qdb_staff.id = qdb_recipient.recipient_id AND qdb_recipient.role = 'aul')
-                OR (qdb_staff.id = qdb_recipient.recipient_id AND qdb_recipient.role = 'assoc'))"""
-        self.cursor.execute(cmd, [unit_id, UL_NAME])
-        recipients.update([r[0] for r in self.cursor.fetchall()])
+            roles = ["aul", "head", "assoc"]
```